lib/eth/key.rb

```diff
@@ -98,9 +98,11 @@ def public_bytes_compressed
       @public_key.compressed
     end
 
+    # Exports the checksummed public address.
+    #
+    # @return [String] compressed address as packed hex prefixed string.
     def address
-      # @TODO Checksummed addresses
-      Utils.public_key_to_address public_bytes
+      Eth::Address.new(Utils.public_key_to_address public_bytes).checksummed
```

Comment:
I wonder if we should return an `Eth::Address` object here (with the risk of breaking backward compatibility). The address class does not serve a great purpose in the `eth` gem; I'm wondering if we return the object, the user can still decide what to do with it, e.g., call `valid?` or get `checksummed`.